### ISSUE TITLE
[Bug] Swift format public modifier breaks extension to class 

### ISSUE THREAD

**Issue by Panajev** [nicklockwood/SwiftFormat#779]
Hello @nicklockwood ,

I have the following case where the latest 0.47.0 Swift Format will break working code (the SGameViewController class, name slightly redacted, is internal/uses default visibility).

Before formatting
```swift
extension SGameViewController {
    public func userContentController(_: WKUserContentController, didReceive message: WKScriptMessage) {
        if (message.name == "game") {
            let body = message.body
            if let body = body as? [String: AnyObject] {
                guard let eventName = body["event"] as? String,
                    let payload = body["payload"] as? [String: String] else {
                    #if !DISTRIBUTION
                        fatalError("incorrect eventName and/or payload [GameViewController]...")
                    #else
                        return
                    #endif
                }

                let event = MessageEvent(event: eventName, payload: payload)
                self.eventHandler.handle(event: event)
            }
        }
    }

    public func messageChannels() -> [String] {
        ["game"]
    }
}
```

After formatting
```swift
public extension SGameViewController {
    func userContentController(_: WKUserContentController, didReceive message: WKScriptMessage) {
        if (message.name == "game") {
            let body = message.body
            if let body = body as? [String: AnyObject] {
                guard let eventName = body["event"] as? String,
                      let payload = body["payload"] as? [String: String] else {
                    #if !DISTRIBUTION
                        fatalError("incorrect eventName and/or payload [GameViewController]...")
                    #else
                        return
                    #endif
                }

                let event = MessageEvent(event: eventName, payload: payload)
                self.eventHandler.handle(event: event)
            }
        }
    }

    func messageChannels() -> [String] {
        ["game"]
    }
}
```

Resulting error:
_extension of internal class cannot be declared public_

Kind Regards,

Goffredo

**Comment by nicklockwood** [comment]
@Panajev I see the problem. The extension had a public function even though the class it's extending was internal. That's permitted, but a public extension isn't. I don't see a workaround for now other than either disabling the `extensionAccessControl` rule or removing the `public` modifier yourself.

**Comment by Panajev** [comment]
Could the extensionAccessControl implementation detect it is going to cause a breakage in this specific scenario (internal extension to an internal class with all public methods declared inside the extension —> do not apply the rule to this extension?) or does it not have visibility of this information (not sure if it could work/how difficult it would be, honest question)?

I guess that disabling this rule for this specific class extension (done on this side with a code comment if possible), removing the public modifier (less likely as CI running formatting and listing would flag a missed out change), disabling the rule altogether, or going along with it and making the class public if it makes sense could be acceptable solutions that require no change for SwiftFormat.

Thanks anyways for the feedback @nick :)!

**Comment by nicklockwood** [comment]
> Could the extensionAccessControl implementation detect it is going to cause a breakage in this specific scenario

Only if the class is defined in the same file as the extension. I could certainly add some logic to handle that case. If the class is defined elsewhere I can't currently detect it, as SwiftFormat only operates on the level of individual files, not whole projects.

**Comment by Panajev** [comment]
Hey @nicklockwood ,

Yes, in this case the class was defined in the same file. It would be awesome to detect it and maybe print something in the consoles to maybe suggest there is a kind of potential code smell. As there could be, the pattern is legal, but it might not be what the dev was actually trying to do so a console hint might be nice (these hints could be added in a "recap" section per file at the end when swiftformat terminates :).

In our case breaking it actually made us think that for the time being we wanted to change the visibility of the class, but there could have been an equally good case not to touch quitea bit of "working" code (we need to refactor again into module, thank you SwiftPM, so I am a lot less fussy about internal vs public at the moment, well I would like to revisit that a lot more in depth, but that is beside the point :)).

Thanks again!

**Comment by nicklockwood** [comment]
@Panajev fixed in 0.47.3 (only for extensions in the same file as the type being extended)

**Comment by Panajev** [comment]
Thanks 🙏
